stable/redis/templates/redis-master-svc.yaml

```diff
@@ -0,0 +1,41 @@
+apiVersion: v1
+kind: Service
+metadata:
+  {{- if .Values.cluster.enabled }}
+  name: {{ template "redis.fullname" . }}-master
+  {{- else }}
+  name: {{ template "redis.fullname" . }}
+  {{- end }}
+  labels:
+    app: {{ template "redis.name" . }}
+    chart: {{ template "redis.chart" . }}
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+  annotations:
+{{- if .Values.master.service.annotations }}
+{{ toYaml .Values.master.service.annotations | indent 4 }}
+{{- end }}
+{{- if .Values.metrics.enabled }}
+{{ toYaml .Values.metrics.targetServiceAnnotations | indent 4 }}
+{{- end }}
+spec:
+  type: {{ .Values.master.serviceType }}
+  {{ if eq .Values.master.serviceType "LoadBalancer" -}} {{ if .Values.master.service.loadBalancerIP -}}
+  loadBalancerIP: {{ .Values.master.service.loadBalancerIP }}
+  {{ end -}}
+  {{- end -}}
+  ports:
+  - name: redis
+    port: 6379
+    targetPort: redis
+  {{- if .Values.metrics.enabled }}
+  - name: metrics
+    port: 9121
+    targetPort: metrics
+  {{- end }}
+  selector:
+    app: {{ template "redis.name" . }}
+    release: "{{ .Release.Name }}"
+    {{- if .Values.cluster.enabled }}
+    role: master
```

Comment:
I would always add the `role: master` label in the statefulset and consequently here as  well